src/groups/mqb/mqbblp/mqbblp_pushstream.h

```diff
@@ -673,7 +660,15 @@ inline unsigned int PushStream::remove(Element* element)
     // remove from the guid
     element->guid().remove(element, e_GUID);
 
-    return element->guid().numElements();
+    if (element->app().d_elements.numElements() == 0) {
```

Comment:
Centralizing this bit of logic as here seems right.  I've traced through the code paths leading up to this and I think this catches all places we need.